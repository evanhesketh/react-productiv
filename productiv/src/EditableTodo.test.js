import React from "react";
import { fireEvent, render } from "@testing-library/react";
import EditableTodo from "./EditableTodo";

const testTodo = {
  title: "Test todo",
  description: "Test description",
  priority: 1
};

describe("EditableTodo", function () {
  it("renders without crashing", function () {
    render(<EditableTodo
      todo={testTodo} />);
  });

  it("matches snapshot on initial render", function () {
    const { container } = render(<EditableTodo todo={testTodo} />);
    expect(container).toMatchSnapshot()
  });

  // it("contains expected information", function () {
  //   const { container } = render(<EditableTodo todo={testTodo} />);
  //   expect(container.querySelector(".EditableTodo")).toContainHTML("Test description");
  //   expect(container.querySelector(".EditableTodo")).toContainHTML("Ultra-Ãœber");
  // });

  it("deleting todo works", function () {
    const mockDelete = jest.fn();
    const { container } = render(<EditableTodo todo={testTodo} remove={mockDelete} />);
    const deleteBtn = container.querySelector('.EditableTodo-delBtn');

    // TODO: test mock hasnt been called and was called after
    fireEvent.click(deleteBtn);

    expect(mockDelete).toHaveBeenCalledTimes(1);
  });

  // TODO: could add test for update - form should appear when edit btn clicked
});