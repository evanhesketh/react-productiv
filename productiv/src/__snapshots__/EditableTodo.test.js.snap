// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTodo matches snapshot on initial render 1`] = `
<DocumentFragment>
  <div
    class="EditableTodo"
  >
    <div
      class="mb-3"
    >
      <div
        class="float-end text-sm-end"
      >
        <button
          class="EditableTodo-toggle btn-link btn btn-sm"
        >
          Edit
        </button>
        <button
          class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
        >
          Del
        </button>
      </div>
      <div
        class="Todo"
      >
        <div>
          <b>
            Test todo
          </b>
           
          <small>
            Priority: 1
          </small>
        </div>
        <div>
          <small>
            Test description
          </small>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
